st2client/st2client/commands/auth.py

```diff
@@ -70,6 +73,84 @@ def run_and_print(self, args, **kwargs):
 
 
+class LoginCommand(resource.ResourceCommand):
+    display_attributes = ['user', 'token', 'expiry']
+
+    def __init__(self, resource, *args, **kwargs):
+
+        kwargs['has_token_opt'] = False
+
+        super(LoginCommand, self).__init__(
+            resource, kwargs.pop('name', 'create'),
+            'Authenticate user, acquire access token, and update CLI config directory',
+            *args, **kwargs)
+
+        self.parser.add_argument('username',
+                                 help='Name of the user to authenticate.')
+
+        self.parser.add_argument('-p', '--password', dest='password',
+                                 help='Password for the user. If password is not provided, '
+                                      'it will be prompted.')
+        self.parser.add_argument('-l', '--ttl', type=int, dest='ttl', default=None,
+                                 help='The life span of the token in seconds. '
+                                      'Max TTL configured by the admin supersedes this.')
+        self.parser.add_argument('-w', '--write-real-password', action='store_true', default=False,
+                                 dest='write_password',
+                                 help='Write the real (plain-text) password to the config file '
+                                      '(default is to omit it')
+
+    def run(self, args, **kwargs):
+
+        if not args.password:
+            args.password = getpass.getpass()
+        instance = self.resource(ttl=args.ttl) if args.ttl else self.resource()
+
+        cli = BaseCLIApp()
+
+        # Determine path to config file
+        try:
+            config_file = cli._get_config_file_path(args)
+        except ValueError:
+            # config file not found in args or in env, defaulting
+            config_file = expanduser('~/.st2/config')
+
+        logging.basicConfig(
+            level=logging.CRITICAL,
+            handlers=[logging.NullHandler]
+        )
+        cli.LOG = logging.getLogger("st2client.base")
+
+        # Retrieve token
+        manager = self.manager.create(instance, auth=(args.username, args.password), **kwargs)
+        cli._cache_auth_token(token_obj=manager)
+
+        # Update existing configuration with new credentials
+        config = ConfigParser()
+        config.read(config_file)
+
+        # Modify config (and optionally populate with password)
+        if 'credentials' not in config:
+            config.add_section('credentials')
+            config['credentials'] == {}
+        config['credentials']['username'] = args.username
+        if args.write_password:
+            config['credentials']['password'] = args.password
+
+        with open(config_file, "w") as cfg_file_out:
+            config.write(cfg_file_out)
+
+        return manager
+
+    def run_and_print(self, args, **kwargs):
+        try:
+            self.run(args, **kwargs)
+            print("Logged in as %s" % args.username)
```

Comment:
Minor style thing - it's usually a good idea to always wrap format string arguments in parenthesis - `print("Logged in as %s" % (args.username))`.

This way it's impossible for thing to break in case someone adds another format value parameter, but forgets to add parenthesis. We have lint, etc. which should, in theory, catch this, but I've still seen it happen before.